unified-installer/astra-unified-installer.sh

```diff
@@ -2714,42 +2868,45 @@ step_apply_trident_operator_patches() {
     local -ra patches=("${_PATCHES_TRIDENT_OPERATOR[@]}")
     local -r patches_len="${#patches[@]}"
 
-    if ! trident_will_be_installed_or_modified && [ "$patches_len" -gt 0 ]; then
+    if [ "$patches_len" -eq 0 ]; then
+        loginfo "* No Astra Trident Operator patches to apply, skipping"
+        return 0
+    elif ! trident_will_be_installed_or_modified; then
         fatal "found $patches_len operator patches (expected 0) despite trident not being installed or modified"
     fi
 
-    if debug_is_on && [ "$patches_len" -gt 0 ]; then
-        local disclaimer="# THIS FILE IS FOR DEBUGGING PURPOSES ONLY. These are the patches applied to the"
-        disclaimer+="${__NEWLINE}# Astra Trident Operator deployment when upgrading the Astra Trident Operator."
-        disclaimer+="${__NEWLINE}"
-        append_lines_to_file "${__GENERATED_PATCHES_TRIDENT_OPERATOR_FILE}" "$disclaimer" "${patches[@]}"
-    fi
+    local disclaimer="# THIS FILE IS FOR DEBUGGING PURPOSES ONLY. These are the patches applied to the"
+    disclaimer+="${__NEWLINE}# Astra Trident Operator deployment when upgrading the Astra Trident Operator."
+    disclaimer+="${__NEWLINE}"
+    append_lines_to_file "${__GENERATED_PATCHES_TRIDENT_OPERATOR_FILE}" "$disclaimer" "${patches[@]}"
```

Comment:
For this, I removed the `debug_is_on` requirement for generating the patch files. We already generate every resource we apply, seemed to make sense to generate this as well (even though we apply the patch from an in-memory reference, not a file -- maybe that should change?).